app/models/host/managed.rb

```diff
@@ -560,7 +560,16 @@ def inherited_attributes
     inherited_attrs = %w{domain_id operatingsystem_id architecture_id compute_resource_id}
     inherited_attrs << "subnet_id" unless compute_provides?(:ip)
     inherited_attrs << "subnet6_id" unless compute_provides?(:ip6)
-    inherited_attrs.concat(%w{medium_id ptable_id pxe_loader}) unless image_build?
+    inherited_attrs.concat(%w{ptable_id pxe_loader}) unless image_build?
+
+    if !image_build? && hostgroup.medium_id.present?
+      if operatingsystem_id.nil? || operatingsystem.medium_ids.include?(hostgroup.medium_id)
+        inherited_attrs.concat(%w{medium_id})
```

Comment:
Nit: it's sufficient to add a single item:
```suggestion
        inherited_attrs << "medium_id"
```

But I'd suggest to keep the `image_build?` part together. Something like:
```ruby
unless image_build?
  inherited_attrs.concat(%w{ptable_id pxe_loader})

  if hostgroup.medium_id.present?
    # ...
  end
end
```

Reply:
thats true. this can be improved.